account_financial_report_qweb/report/open_items.py

```diff
@@ -428,27 +425,20 @@ def _get_line_sub_query_move_lines(self,
             LEFT JOIN
                 account_partial_reconcile pr
                     ON ml.balance > 0 AND pr.debit_move_id = ml.id
-            LEFT JOIN
-                account_move_line ml_future
-                    ON ml.balance > 0 AND pr.credit_move_id = ml_future.id
-                    AND ml_future.date > %s
             LEFT JOIN
                 account_move_line ml_past
                     ON ml.balance > 0 AND pr.credit_move_id = ml_past.id
                     AND ml_past.date <= %s
         """
         sub_query += """
+            LEFT JOIN account_full_reconcile afr ON afr.id = ml.full_reconcile_id
             WHERE
                 ra.report_id = %s
+            AND ml.full_reconcile_id IS NULL OR afr.create_date >= %s
             GROUP BY
                 ml.id,
                 ml.balance,
                 ml.amount_currency
```

Comment:
IMO, this GROUP BY can be removed. GROUP BY id does not make much sense to me, and the id should be unique across the two parts of the union.

Reply:
Hm, no I retract this. There are these LEFT JOIN that repeat the id.